lib/_emerge/Package.py

```diff
@@ -314,6 +314,10 @@ def _validate_deps(self):
         eapi = self.eapi
         dep_eapi = eapi
         dep_valid_flag = self.iuse.is_valid_flag
+
+        if not force and not self.root_config.settings.repositories[self.repo].volatile:
+            return
+
```

Comment:
The _validate_deps method has parse_soname_deps calls for PROVIDES and REQUIRES that we don't want to skip for binary packages.